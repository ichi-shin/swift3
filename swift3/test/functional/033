#!/bin/bash

# Bucket logging

. ./common

# FIXME: this works only on my environment

S3USER=tester

_s3_put /bucket -H 'x-amz-acl: private'
echo '<?xml version="1.0" encoding="UTF-8"?>
<BucketLoggingStatus xmlns="http://s3.amazonaws.com/doc/2006-03-01/">
  <LoggingEnabled>
    <TargetBucket>bucket2</TargetBucket>
    <TargetPrefix>log-</TargetPrefix>
  </LoggingEnabled>
</BucketLoggingStatus>' > $tmp.logging.xml
_s3_put /bucket2 -H 'x-amz-acl: log-delivery-write'
_s3_put /bucket?logging -T $tmp.logging.xml

mkdir -p /tmp/swift/log2
cp /tmp/swift/log/proxy.log /tmp/swift/log2
sleep 3
. ~/git/swifttest/.venv/bin/activate
swift3-log-delivery ~/git/swift3/etc/swift3-log-delivery.conf -v -o
deactivate

for log in $(_sw_get /bucket2); do
    _s3_get /bucket2/$log
done
