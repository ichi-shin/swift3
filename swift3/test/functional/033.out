QA output created by 033
> s3curl --id tester -- -X PUT -H x-amz-acl: private http://SWIFT_HOST/bucket... 200
> s3curl --id tester -- -X PUT -H x-amz-acl: log-delivery-write http://SWIFT_HOST/bucket2... 200
> s3curl --id tester -- -X PUT -T /TMP.logging.xml http://SWIFT_HOST/bucket?logging... 200
swift3-log-delivery: Adding required filter dlo to pipeline at position 1
swift3-log-delivery: Adding required filter gatekeeper to pipeline at position 1
swift3-log-delivery: Pipeline was modified. New pipeline is "catch_errors gatekeeper dlo cache proxy-server".
swift3-log-delivery: Uploading logs
swift3-log-delivery: /tmp/swift/log2/proxy.log
swift3-log-delivery: Processing log: /tmp/swift/log2/proxy.log
swift3-log-delivery: Uploading logs complete (0.00 minutes)
swift3-log-delivery: Exited
> curl -X GET http://SWIFT_HOST/v1/TENANT/bucket2... 204
